share/p2p/discovery/metrics.go

```diff
@@ -53,43 +53,43 @@ func (d *Discovery) WithMetrics() error {
 }
 
 func initMetrics(d *Discovery) (*metrics, error) {
-	peersAmount, err := meter.Int64ObservableGauge("discovery_amount_of_peers",
+	peersAmount, err := meter.Int64ObservableGauge(fmt.Sprintf("%s_discovery_amount_of_peers", d.tag),
```

Comment:
Nit: could be just:
```
d.tag + "_discovery_amount_of_peers"
```
I personally find it cognitively harder to parse template.